src/W3GReplay.ts

```diff
@@ -148,6 +150,13 @@ class W3GReplay extends ReplayParser {
     }
 
     handleActionBlock (action: ActionBlock, currentPlayer: Player) {
+        this.playerActionTracker[currentPlayer.id] = this.playerActionTracker[currentPlayer.id] || []
+        this.playerActionTracker[currentPlayer.id].push(action)
+
+        if (action.itemId && (action.itemId.value === 'tert' || action.itemId.value === 'tret')) {
+            console.log('retraining action detected!', action)
```

Comment:
noticed some debug logs here

Reply:
Yes, currently trying to assert whether or not its possible to detect the specific hero that uses the item. Looks like it isnt as of my current investigations.

As a fallback, we could use the same method as in Julas w3g parser to detect tome of retraining usage.

Reply:
my project built on this parser aims to do more in depth detection like this - https://github.com/jblanchette/wc3v

it's still heavy wip and pre-release state.  but it turns out you can track most things